src/Sentry/Internal/DefaultSentryStructuredLogger.cs

```diff
@@ -0,0 +1,78 @@
+using Sentry.Extensibility;
+using Sentry.Infrastructure;
+using Sentry.Protocol.Envelopes;
+
+namespace Sentry.Internal;
+
+internal sealed class DefaultSentryStructuredLogger : SentryStructuredLogger
+{
+    private readonly IHub _hub;
+    private readonly SentryOptions _options;
+    private readonly ISystemClock _clock;
+
+    internal DefaultSentryStructuredLogger(IHub hub, SentryOptions options, ISystemClock clock)
+    {
+        Debug.Assert(options is { Experimental.EnableLogs: true });
+
+        _hub = hub;
+        _options = options;
+        _clock = clock;
+    }
+
+    private protected override void CaptureLog(SentryLogLevel level, string template, object[]? parameters, Action<SentryLog>? configureLog)
+    {
+        var timestamp = _clock.GetUtcNow();
+        var traceHeader = _hub.GetTraceHeader() ?? SentryTraceHeader.Empty;
+
+        string message;
+        try
+        {
+            message = string.Format(CultureInfo.InvariantCulture, template, parameters ?? []);
+        }
+        catch (FormatException e)
+        {
+            _options.DiagnosticLogger?.LogError(e, "Template string does not match the provided argument. The Log will be dropped.");
+            return;
+        }
+
+        SentryLog log = new(timestamp, traceHeader.TraceId, level, message)
+        {
+            Template = template,
+            Parameters = ImmutableArray.Create(parameters),
+            ParentSpanId = traceHeader.SpanId,
+        };
+
+        try
+        {
+            configureLog?.Invoke(log);
```

Comment:
I find it odd that we have this callback to be invoked directly when capturing a log.

What are the use cases that we'll call explicitly: `Log(a,b,c,d, (log) => {})` when `log` contains the fields I anyway just provided to the `Log()` siuch as `a,b,c,d`?)

The use case for a callback when the user isn't calling the `Log()` method directly makes a lot of sense to me, and we have that below with `beforeLog` but having both is odd.